rviz_default_plugins/src/rviz_default_plugins/displays/marker/markers/mesh_resource_marker.cpp

```diff
@@ -228,22 +235,38 @@ MeshResourceMarker::updateMaterialColor(const MarkerBase::MarkerConstSharedPtr &
 
   Ogre::SceneBlendType blending;
   bool depth_write;
+  bool tinting = new_message->mesh_use_embedded_materials;
 
   rviz_rendering::MaterialManager::enableAlphaBlending(blending, depth_write, a);
 
   if (new_message->mesh_use_embedded_materials && r == 0 && g == 0 && b == 0 && a == 0) {
     blending = Ogre::SBT_REPLACE;
     depth_write = true;
-    r = 1; g = 1; b = 1; a = 1;
   }
 
   for (auto & material : this->materials_) {
-    Ogre::Technique * technique = material->getTechnique(0);
-    technique->setAmbient(r * 0.5f, g * 0.5f, b * 0.5f);
-    technique->setDiffuse(r, g, b, a);
-    technique->setSceneBlending(blending);
-    technique->setDepthWriteEnabled(depth_write);
-    technique->setLightingEnabled(true);
+      Ogre::Technique* technique = material->getTechnique(0);
+      Ogre::Pass* pass0 = technique->getPass(0);
+      Ogre::Pass* passT = technique->getPass(technique->getNumPasses() - 1);
+      if (tinting)
+      {
```

Comment:
Actually, CI is complaining about the style here.  This should be:

```suggestion
      if (tinting) {
```